app.py

```diff
@@ -0,0 +1,437 @@
+        headers=headers["prod"],
+        data=message_payload,
+    )
+
+    # if a refund, post a new record to knack transactions table
+    if payment_status == "refunded":
+        knack_payload = get_knack_refund_payload(
+            "prod",
+            payment_status,
+            payment_amount,
+            knack_invoice,
+            today_date,
+            knack_record_id,
+        )
+        knack_response = requests.post(
+            f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID}/records/",
+            headers=headers["prod"],
+            data=knack_payload,
+        )
+    # otherwise, update existing record payment status on transactions table
+    else:
+        knack_payload = get_knack_payload("prod", payment_status, today_date)
+        if payment_status == "successful":
+            # if this was a successful payment we also need to update reservation record
+            update_parent_reservation("prod", knack_record_id, today_date)
+        knack_response = requests.put(
+            f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID}/records/{knack_record_id}",
+            headers=headers["prod"],
+            data=knack_payload,
+        )
+    if knack_response.status_code == 200:
+        return "Payment status updated", knack_response.status_code
+    # if unsuccessful, return knack's status response as response
+    return knack_response.text, knack_response.status_code
+
+
+@app.route("/citybase_postback_uat", methods=["POST"])
+def handle_postback_uat():
+    today_date = datetime.now().strftime("%m/%d/%Y %H:%M")
+    citybase_data = request.get_json()
+
+    knack_record_id = get_knack_record_id(citybase_data)
+    # if knack_record_id is not a string, then it is an error tuple
+    if not isinstance(knack_record_id, str):
+        return knack_record_id
+
+    knack_invoice = get_knack_invoice(citybase_data)
+    # if knack_record_id is not a string, then it is an error tuple
+    if not isinstance(knack_invoice, str):
+        return knack_invoice
+
+    payment_status = citybase_data["data"]["status"]
+    payment_amount = citybase_data["data"]["total_amount"]
+    citybase_id = citybase_data["data"]["id"]
+
+    message_payload = create_message_json(
+        "uat", citybase_id, today_date, knack_invoice, payment_status
+    )
+    requests.post(
+        f"{KNACK_API_URL}{MESSAGES_OBJECT_ID_UAT}/records/",
+        headers=headers["uat"],
+        data=message_payload,
+    )
+
+    # if a refund, post a new record to knack transactions table
+    if payment_status == "refunded":
+        knack_payload = get_knack_refund_payload(
+            "uat",
+            payment_status,
+            payment_amount,
+            knack_invoice,
+            today_date,
+            knack_record_id,
+        )
+        knack_response = requests.post(
+            f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID_UAT}/records/",
+            headers=headers["uat"],
+            data=knack_payload,
+        )
```

Comment:
Not suggesting any changes but I wonder how this would be different if we used the Knack forms API to submit these records? It could handle some of the logic we have here. John pointed it out that we use it in knack-services [here](https://github.com/cityofaustin/atd-knack-services/blob/8adedf3e4a8bb7b47a464d6048328e84a52041c4/services/sr_asset_assign.py#L137).